### DISCUSSION TITLE
Generated wallets for user are different in Web SDK and React Native SDK

### DISCUSSION THREAD

**Discussion by amatevosyan-github** [Web3Auth/Web3Auth#1114] [Help]
Does web3auth generates different wallets for the same social account if **different SDKs** was used with the **same clientId** ? I created a web app and react-native app and got 2 different wallets generated. when connected via the same Google account. Question is - is it intended that way, if so, is there a workaround?

**Comment by sebastian-baier** [comment]
Hi did you find a solution?
I have the exact same problem

**Reply by amatevosyan-github** [reply]
Nope :(

**Comment by yashovardhan** [comment]
Hey @amatevosyan-github 

Can you tell me which SDKs are you using? Web3Auth should return the same key if you're using the same verifier and client id & network across any of the SDKs. Please note that even while using the same google account with custom authentication (another verifier) the keys will change, however, if the configs are same across different SDKs, the resultant keys will be the same.

You can check our examples and test it out:
- https://github.com/Web3Auth/examples/tree/main/react-native/rn-expo-example
- https://github.com/Web3Auth/examples/tree/main/web-modal-sdk/evm/react-evm-web3auth-example

You can test in our hosted examples here as well:
https://main.dmmzgprvd8gqq.amplifyapp.com/
https://appetize.io/app/kakxmpaihaoflf4kp43qfm5uz4?device=pixel4&osVersion=11.0&scale=75

**Reply by sebastian-baier** [reply]
thanks, works now for me
looks like it just works with the mainnet
the other networks are not working for me

**Reply by yashovardhan** [reply]
Yes the network parameter, if is the same across it should work.

**Reply by sebastian-baier** [reply]
for me it just worked with mainnet on both
what I also tested was: testnet on both and cyan on both (of course I switched the clientId to the matching networks)
But this setups did not work for me

**Comment by ankushagarwal** [comment]
@yashovardhan : I had a question regarding this

> Web3Auth should return the same key if you're using the same verifier and client id & network across any of the SDKs.

Let's say I have a verifier named `ankush-custom-verifier` and I accidentally delete it, do I lose all the keys associated with the verifier?
Is there any way to recover it? Is it possible to recreate the same verifier with the same params (and recover the keys)?
